k4Reco/Overlay/components/OverlayTiming.cpp

```diff
@@ -0,0 +1,379 @@
+      for (auto& file : m_inputFileNames.value()) {
+        err += " " + file[0];
+      }
+      error() << err << endmsg;
+      return StatusCode::FAILURE;
+    }
+  }
+
+  if (std::any_of(m_bkgEvents->m_totalNumberOfEvents.begin(), m_bkgEvents->m_totalNumberOfEvents.end(),
+                  [this](const int& val) { return this->m_startWithBackgroundEvent >= val; })) {
+    throw GaudiException("StartWithBackgroundEvent is larger than the number of events in the background files", name(),
+                         StatusCode::FAILURE);
+    return StatusCode::FAILURE;
+  }
+
+  if (m_Noverlay.empty()) {
+    info() << "Using the default number of overlay events (1) for each group, since none was specified with "
+              "NumberBackground "
+           << endmsg;
+    m_Noverlay = std::vector<double>(m_bkgEvents->size(), 1);
+  }
+
+  if (m_Poisson.empty()) {
+    info() << "Using the default overlay mode (no Poission distribution) for each group, since none was specified with "
+              "Poisson_random_Noverlay"
+           << endmsg;
+    m_Poisson = std::vector<bool>(m_bkgEvents->size(), false);
+  }
+
+  return StatusCode::SUCCESS;
+}
+
+retType OverlayTiming::operator()(const edm4hep::EventHeaderCollection&                           headers,
+                                  const edm4hep::MCParticleCollection&                            particles,
+                                  const std::vector<const edm4hep::SimTrackerHitCollection*>&     simTrackerHits,
+                                  const std::vector<const edm4hep::SimCalorimeterHitCollection*>& simCaloHits) const {
+  const auto seed = m_uidSvc->getUniqueID(headers[0].getEventNumber(), headers[0].getRunNumber(), this->name());
+  m_engine.seed(seed);
+
+  // Output collections
+  auto oparticles       = edm4hep::MCParticleCollection();
+  auto osimTrackerHits  = std::vector<edm4hep::SimTrackerHitCollection>();
+  auto osimCaloHits     = std::vector<edm4hep::SimCalorimeterHitCollection>();
+  auto ocaloHitContribs = std::vector<edm4hep::CaloHitContributionCollection>();
+  for (size_t i = 0; i < simCaloHits.size(); ++i) {
+    ocaloHitContribs.emplace_back(edm4hep::CaloHitContributionCollection());
+  }
+
+  // Copy MCParticles for physics event into a new collection
+  for (auto&& part : particles) {
+    oparticles->push_back(part.clone(false));
+  }
+  // Fix relations to point to the new particles
+  for (size_t i = 0; i < particles.size(); ++i) {
+    for (auto& parent : particles[i].getParents()) {
+      oparticles[i].addToParents(oparticles[parent.getObjectID().index]);
+    }
+    for (auto& daughter : particles[i].getDaughters()) {
+      oparticles[i].addToDaughters(oparticles[daughter.getObjectID().index]);
+    }
+  }
+
+  // Copy the SimTrackerHits and crop them
+  for (size_t i = 0; i < simTrackerHits.size(); ++i) {
+    auto& coll                   = simTrackerHits[i];
+    auto  name                   = m_inputLocations[2][i].key();
```

Comment:
```suggestion
    auto  name                   = m_inputLocations[TRACKERHIT_INDEX_POSITION][i].key();
```

Reply:
Where does `m_inputLocations` come from? Inherited from MultiTransformer?

Reply:
Not used anymore, now `inputLocations()` is used.